addons/base_vat/models/res_partner.py

```diff
@@ -224,6 +224,25 @@ def check_vat_al(self, vat):
             return True
         return False
 
+    __check_tin_hu_companies_re = re.compile(r'^8\d{9}$')
+    __check_tin_hu_individual_re = re.compile(r'^\d{8}-[1-5]-\d{2}$')
+
+    def check_vat_hu(self, vat):
+        """
+            Check Hungary VAT number that can be for example 'HU12345676 or 'xxxxxxxx-y-zz' or '8xxxxxxxxy'
+            - For xxxxxxxx-y-zz, 'x' can be any number, 'y' is a number between 1 and 5 depending on the person and the 'zz'
+              is used for region code.
+            - 8xxxxxxxxy, Tin number for individual, it has to start with an 8 and finish with the check digit
+        """
+        companies = self.__check_tin_hu_companies_re.match(vat)
+        if companies:
+            return True
+        individual = self.__check_tin_hu_individual_re.match(vat)
+        if individual:
+            return True
+        # Check the vat number
+        return getattr(stdnum.util.get_cc_module('hu', 'vat'), 'is_valid', None)
```

Comment:
You should call the function, otherwise you just return the function and we only check if it exists (which is the case), so it always returns something truethy